torch/custom_class.h

```diff
@@ -263,11 +298,46 @@ class class_ {
 
  private:
   template <typename Func>
-  void defineMethod(std::string name, Func func, std::string doc_string = "") {
+  void defineMethod(
+      std::string name,
+      Func func,
+      std::string doc_string = "",
+      std::initializer_list<arg> default_args = {}) {
     auto qualMethodName = qualClassName + "." + name;
     auto schema = c10::inferFunctionSchemaSingleReturn<Func>(std::move(name), "");
```

Comment:
I think `inferFunctionSchemaSingleReturn` is only used by TorchBind, could we merge all the logic below related to schema construction into `inferFunctionSchemaSingleReturn`?

Reply:
I would prefer to keep schema inference + attaching default args separatel.